Authentication.md

```diff
@@ -1,9 +1,10 @@
 # Authentication
 The java-sdk-core project supports the following types of authentication:
 - Basic Authentication
-- Bearer Token 
```

Comment:
Some of the changes to this document are to track more closely the Go core version.